desc/optimize/least_squares.py

```diff
@@ -178,7 +178,7 @@ def lsqtr(  # noqa: C901
     f = fun(x, *args)
     nfev += 1
     cost = 0.5 * jnp.dot(f, f)
-    J = jac(x, *args)
+    J = jac(x, *args).block_until_ready()  # FIXME: block is needed for jaxify util
```

Comment:
did you figure out what was going on here? This feels like a kludge, and an incomplete one at that (ie wouldn't we need to do similar things for all the other optimizers?)

Reply:
I am still not sure why this is needed. Last time I tested on all the optimizers (a couple of months ago) it was only an issue for our custom least-squares optimizer. The SciPy ones and our augmented-Lagrangian were fine. 

I will try testing again, but I might need help if we want to avoid this kludge.